src/wp-includes/blocks.php

```diff
@@ -2139,11 +2139,6 @@ function build_comment_query_vars_from_block( $block ) {
 					$comment_args['paged'] = $max_num_pages;
 				}
 			}
-			// Set the `cpage` query var to ensure the previous and next pagination links are correct
-			// when inheriting the Discussion Settings.
-			if ( 0 === $page && isset( $comment_args['paged'] ) && $comment_args['paged'] > 0 ) {
-				set_query_var( 'cpage', $comment_args['paged'] );
-			}
```

Comment:
I'm afraid removing this will cause another bug. This code was introduced in [this pull request](https://github.com/WordPress/gutenberg/pull/39227) for a reason. More context can be found there and in [this comment](https://github.com/WordPress/gutenberg/pull/39227#issuecomment-1062903308) that code was suggested. We might want to revisit other potential solutions to that bug. There are a few suggestions on that discussion, although it is a bit old.

Reply:
That might be because of an off-by-one error. Patching the query should not be considered a fix for the original error.